### PR TITLE
step 3: handling JSON responses in favourite pokemon button

### PR DESCRIPTION
**Pull request by leafgecko**
use the fetch() function to request comments data
server responds with some hard-coded JSON data
JavaScript should add these comments to the page on "Pokemon" button

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (fanwgwg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Can we make this a `final` list that's initialized at declaration? So far I don't see a need for re-assigning this list. Also, a better name would be `pokemonList` to reflect its type.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (fanwgwg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Same for Gson, as it's used in each get request, I would prefer to make a class level `private final` field. Such that we don't need to instantiate a new Gson every time a new request received.

File: portfolio/src/main/webapp/script.js
Comment (fanwgwg) [inline portfolio/src/main/webapp/script.js]: nit: This naming isn't quite meaningful, shall we name it `showFavouritePokemon` intead?

**Review by fanwgwg — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (leafgecko) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: done!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (leafgecko) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: done, thank you! To double-check, if it an engineering practice that for every variable I initialized, it should be automatically set to private if it's meant to only be used within that class? 

2nd qn: If you can initialize things in class level then how do you decide what to initialize in the init function?

File: portfolio/src/main/webapp/script.js
Comment (leafgecko) [reply portfolio/src/main/webapp/script.js]: true, changed! thank you

**Comment by leafgecko** [comment]
fixed some items, let me know if this works thank you!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (fanwgwg) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: 1. Yes it must be private.
2. There may be things you don't know at time of declaration, for example, instantiation of A depends on instantiation of B. For example, for your pokemon list here, now we're putting in some hardcoded pokenmons. However, in real codebase, there must be a data source that's providing the stored pokemons, and we can only cosntruct such list after retrieving the pokemons from the data source. In such case, we won't be able to fill in every data before some instantiation and logic with other objects' instantiations.

Also, I do think that you should move the logic of adding elements into the pokenmon list into `init`. Like the following:
``` 
private final List<String> pokemonList = new ArrayList<>();

@override
void init() {
   pokemonList.add(...);
   pokemonList.add(...);
   pokemonList.add(...);
}
```

Comment (fanwgwg) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: But to unblock you I'll approve this first, feel free to make this change in a later PR, or make the change before merge :)

**Review by fanwgwg — APPROVED** [review]

**Pull request merged**
